pkg/cli/logfile/logfile.go

```diff
@@ -107,9 +124,14 @@ func Hook() error {
 		<-exited
 	} else {
 		rec := newRecorder(logFile, 0, 0, cmd.Path, cmd.Args)
+		finishedChan := make(chan struct{})
```

Comment:
Move this into the `recorder` function instead so that it can be shared between both of the functions